V2023 1

Samet-MohamedAmin opened the pull request:
Course changes:
- Introduce workshop env
- Change containers introduction
- Version updates
- Add chapter best practices
- Add chapter CRDs
- Add chapter concusion 

Workshop Changes:
- Version updates
- Use `kind` instead of `minikube`
- Update `kind-install` workshop
   - Install `kind`
   - Create/delete cluster
   - Get to know `kubeconfig`
- Add workshop `k8s-first-deloyments`: 
   -  Minimal workaround to create a functioning app
   - Create pod, service and namespace
   - Imperative approach to get ease with `kubectl`
   - Declarative approach
   - Debug
   - Install kubernetes dashboard
- Update workshop `k8s-first-deloyments`:
   - Create first deployment
   - Manual scale
   - RollingUpdate
   - Install ingress controller
   - Use ingress to expose service
   - Autoscaling with stress test image
- Update workshop `k8s-secrets-cm-volumes`
   - `InitContainer` introduction
   - `PeristentVolume` introduction

Samet-MohamedAmin committed:
update
update copyright year
add chapter cloud-introduction
update containers-introduction content

Samet-MohamedAmin committed:
add CRDs chapter

Samet-MohamedAmin committed:
update small changes
uppercase elements
use batch/v1 for cronjob
separate long chapters

Samet-MohamedAmin committed:
update kind version

Samet-MohamedAmin committed:
add cloud introduction et k8s best practices chapters

Samet-MohamedAmin committed:
update kind-installation chapter

Samet-MohamedAmin committed:
update

Samet-MohamedAmin committed:
update

Samet-MohamedAmin committed:
add section kubernetes components
It is meant to give global vue of kuberentes componentes

Samet-MohamedAmin committed:
update

Samet-MohamedAmin committed:
add Problématiques section

Samet-MohamedAmin committed:
add Control plane componentes & kubeconfig

Samet-MohamedAmin committed:
update modules order

Samet-MohamedAmin committed:
update

Samet-MohamedAmin committed:
add registry to image name

Samet-MohamedAmin committed:
update k8s dashboard link

Samet-MohamedAmin committed:
update speraate chapters deployment and pod
add chapter first pod. attendees will focus on learning how to interact with kubernetes with fewer objects
add chapter frst deployment. it should contain deployment, ingress and rollout updates

Samet-MohamedAmin committed:
update

Samet-MohamedAmin committed:
update

Samet-MohamedAmin committed:
update

Samet-MohamedAmin committed:
update dashboard

Samet-MohamedAmin committed:
update
nginx kind cluster
add hpa section

Samet-MohamedAmin committed:
update resources

Samet-MohamedAmin committed:
use cluster-admin role

Samet-MohamedAmin committed:
update helm installation

Samet-MohamedAmin committed:
add tp introduction
explain how to use tp environment

Samet-MohamedAmin committed:
fix name

Samet-MohamedAmin committed:
small fixes

Samet-MohamedAmin committed:
small changes

Samet-MohamedAmin committed:
update

Samet-MohamedAmin committed:
update

Samet-MohamedAmin committed:
update base image
small changes

Samet-MohamedAmin committed:
update

Samet-MohamedAmin committed:
small changes

Samet-MohamedAmin committed:
update adjust values

Samet-MohamedAmin committed:
update

Samet-MohamedAmin committed:
add secret creation for service account

Samet-MohamedAmin committed:
add volumes and init containers

Samet-MohamedAmin committed:
update conclusion

Samet-MohamedAmin committed:
fix port

Samet-MohamedAmin committed:
fix spaces

Samet-MohamedAmin committed:
add kubernetes deploy options

Samet-MohamedAmin committed:
add chapter autoscaling

Samet-MohamedAmin committed:
remove dashboard

Samet-MohamedAmin committed:
update
remove section hpa
ajouter une chapitre à vous de faire

Samet-MohamedAmin committed:
add chapter dashboard

Samet-MohamedAmin committed:
add dashboard section to rbac

Samet-MohamedAmin committed:
update

rguichard commented on cours/containers/kubernetes/conclusion.fr.md:
```diff
@@ -1,3 +1,23 @@
+
+### L'API de Kubernetes est ouvertes
+- Kubernetes a plusieurs composants
+- Il suffit de respecter la description des api dans kubernetes pour
```
pour ?

rguichard commented on cours/containers/kubernetes/conclusion.fr.md:
```diff
@@ -1,3 +1,23 @@
+No vendor locking
+
+### Standarisation des ressources
+Tous les composants nécessaires pour déployer les applications dans kubernetes peuvent être définis sous forme des manifests et tout passe par kubernetes.
```
Repréciser la phrase, je ne comprends pas le sens

rguichard commented on cours/containers/kubernetes/conclusion.fr.md:
```diff
@@ -1,3 +1,23 @@
+
+
+### Plus de contrôle sur l'état du système
+Avec kubernetnes, on définit l'état désirée sous forme yaml et kuberentes s'occupe de converger l'état actuelle vers l'état désirée.
```
- état désiré
- Kubernetes

rguichard commented on cours/containers/kubernetes/k8s-architecture.fr.md:
```diff
@@ -7,17 +7,23 @@
-### Kubernetes : Composants du Control Plane
+### Kubernetes : Composants
+- Kubernetes fonctionne en cluster
+- Un cluster un groupement des noeuds (VMs)
```
pas forcément des VM

rguichard commented on cours/containers/kubernetes/k8s-architecture.fr.md:
```diff
@@ -7,17 +7,23 @@
+### Kubernetes : Composants
+- Kubernetes fonctionne en cluster
+- Un cluster un groupement des noeuds (VMs)
+- Les noeuds ont deux rôles:
```
"ont des rôles". On peut avoir les deux, ou seulement l'un des deux.

rguichard commented on cours/containers/kubernetes/k8s-best-practices.fr.md:
```diff
@@ -0,0 +1,27 @@
+    - la version `latest` n'est pas la plus stable 
+
+
+### Specification des ressources
```
Spécification

rguichard commented on cours/containers/kubernetes/k8s-best-practices.fr.md:
```diff
@@ -0,0 +1,27 @@
+
+
+### L'utilisation des policies
+Chaque requête vers l'api de kubernetes passe par l'`admission controller`.
```
Kubernetes

rguichard commented on cours/containers/kubernetes/k8s-best-practices.fr.md:
```diff
@@ -0,0 +1,27 @@
+
+### L'utilisation des policies
+Chaque requête vers l'api de kubernetes passe par l'`admission controller`.
+La validation des objets à déployer passe se fait au de l'`admission controller`.
```
il manque des mots

rguichard commented on cours/containers/kubernetes/k8s-crds.fr.md:
```diff
@@ -0,0 +1,57 @@
+### Kuberentes est extensible
+- L'architecture de kubernetes est ouverte
+- Kubernetes de se nature est extensible
+- On peut définir d'autres ressources dans l'api-service pour les utiliser après
```
les utiliser plus tard

rguichard commented on cours/containers/kubernetes/k8s-crds.fr.md:
```diff
@@ -0,0 +1,57 @@
+- L'architecture de kubernetes est ouverte
+- Kubernetes de se nature est extensible
+- On peut définir d'autres ressources dans l'api-service pour les utiliser après
+- Potentiellement, l'uilisatuer de kubernetes va rencontrer des CRDs 
```
- utilisateur
- Kubernetes

rguichard commented on cours/containers/kubernetes/k8s-crds.fr.md:
```diff
@@ -0,0 +1,57 @@
+
+
+### Custom Resource et Custom Resource Definition
+- La différence entre CR et CRD est la même différence entre le pod qui tourne et le manifest yaml du pod
```
il faut être plus précis que ça.

Samet-MohamedAmin replied:
Il faut parler des controlleurs ?

rguichard commented on cours/containers/kubernetes/k8s-crds.fr.md:
```diff
@@ -0,0 +1,57 @@
+
+
+### Les controlleurs de kubernetes
+- Pour chaque objet dans kubernetes il y a un controlleur qui s'assure du bon fonctionnement de l'objet.
```
- Kubernetes
- ce n'est pas vrai pour les pods, les secrets, configmap etc

rguichard commented on cours/containers/kubernetes/k8s-where-how.fr.md:
```diff
@@ -129,3 +128,22 @@ Infrastructure immuable
+### RKE
+- Facile à utiliser
+- Offert par Rancher
+- configuration des cluster avec YAML
```
- Configuration
- clusters

rguichard commented on cours/containers/kubernetes/k8s-where-how.fr.md:
```diff
@@ -129,3 +128,22 @@ Infrastructure immuable
+- examples:
+  - [k3s](https://k3s.io/)
+  - [k0s](https://k0sproject.io/)
+- Distrubitons kuebrnetes minimales
```
- Distributions
- Kubernetes
- minimalistes

rguichard commented on cours/containers/kubernetes/k8s-where-how.fr.md:
```diff
@@ -129,3 +128,22 @@ Infrastructure immuable
+  - [k3s](https://k3s.io/)
+  - [k0s](https://k0sproject.io/)
+- Distrubitons kuebrnetes minimales
+- Support de l'architecture ARM pour les cartes électironiques d'IOT
```
électroniques

rguichard commented on cours/general/tp-introduction.fr.md:
```diff
@@ -0,0 +1,41 @@
+
+- Un terminal 
+
+- Un editeur de texte
```
éditeur

rguichard commented on tp/containers/k8s-first-deloyments.fr.md:
```diff
@@ -0,0 +1,223 @@
+- Kubectl
+- Wget
+
+Le cluster de type kind doit avoir une configuration simulaire:
```
similaire

rguichard reviewed (changes requested):
- Dans les TP, parfois tu utilises l'indicatif "-er", parfois l'impératif "-ez", il faut en choisir un des - deux
- Kubernetes prend toujours une majuscule

(je n'ai pas encore review les TP)

Samet-MohamedAmin committed:
update fix reviews de Romain

Samet-MohamedAmin committed:
remove topology key as it is deprecated

Samet-MohamedAmin committed:
add cource plan

Samet-MohamedAmin committed:
update chapters order

Samet-MohamedAmin committed:
update

Samet-MohamedAmin committed:
language check

rguichard commented on cours/containers/docker-bsr.fr.md:
```diff
@@ -197,3 +240,4 @@ docker import mon-image.tar backup/mon-conteneur
 
 - Contrôle au niveau conteneurs
 
+- Besoin d'un outil gestion des contenurs à grand échèle
```
outil de gestion des conteneurs à grande échelle

rguichard commented on cours/containers/kubernetes/conclusion.fr.md:
```diff
@@ -1,3 +1,23 @@
+
+### L'API de Kubernetes est ouvertes
+- Kubernetes a plusieurs composants
+- Il suffit de respecter la description des APIs dans Kubernetes pour ajouter un intégrer un nouveau composant
```
ajouter ou intégrer

rguichard commented on cours/containers/kubernetes/conclusion.fr.md:
```diff
@@ -1,3 +1,23 @@
+
+
+### Plus de contrôle sur l'état du système
+Avec Kubernetes, on définit l'état désiré sous forme yaml et Kuberentes s'occupe de converger l'état actuel vers l'état désiré.
```
Kubernetes

Samet-MohamedAmin committed:
fix

Samet-MohamedAmin committed:
fix

Samet-MohamedAmin committed:
fix

Samet-MohamedAmin committed:
update

Samet-MohamedAmin committed:
update

Samet-MohamedAmin committed:
fix
utiliser l'impératif "-ez" au lieu de l'indicatif "-er"

Samet-MohamedAmin committed:
update course order

Samet-MohamedAmin committed:
fix

Samet-MohamedAmin committed:
fix

Samet-MohamedAmin committed:
update

Samet-MohamedAmin committed:
Update conclusion.fr.md

rguichard commented on tp/containers/k8s-rbac.fr.md:
```diff
@@ -170,10 +170,19 @@ d'hériter des droits du ServiceAccount.
 ---
 apiVersion: rbac.authorization.k8s.io/v1
-kind: RoleBinding
+kind: RoleBindinger
```
typo

Samet-MohamedAmin committed:
fix

Samet-MohamedAmin committed:
remove init container part

Samet-MohamedAmin committed:
fix
utiliser l'impératif

Samet-MohamedAmin committed:
update

Samet-MohamedAmin committed:
fix
bad pdf text formatting with many spaces

Samet-MohamedAmin committed:
fix spaces

Samet-MohamedAmin committed:
update
install calico

Samet-MohamedAmin committed:
update ingress controller version

Samet-MohamedAmin committed:
update spaces

Samet-MohamedAmin committed:
update
apiVersion: autoscaling/v2

Samet-MohamedAmin committed:
update
remove PSP - deprecated

Samet-MohamedAmin committed:
update
add chapter k8s introduction
add diagrams: containers idea, base image, simplified k8s arch, pod
add commands to docker bsr: build & rmi
rename tp k8s-first-deployment.fr.md
add serviceaccount config to access k8s dashboard to rbac tp
update chapters order
